js/views/roomlistview.js

```diff
@@ -191,19 +187,23 @@
 				return;
 			}
 
-			//If user is in that room, it should leave that room first.
-			if (this.model.get('active')) {
-				OCA.SpreedMe.app.connection.leaveCurrentCall(true);
-				OC.Util.History.pushState({}, OC.generateUrl('/apps/spreed'));
-			}
-
+			this.exitingActiveRoom();
 			this.$el.slideUp();
 
 			$.ajax({
 				url: OC.linkToOCS('apps/spreed/api/v1/room', 2) + this.model.get('token'),
 				type: 'DELETE'
 			});
 		},
+		exitingActiveRoom: function() {
```

Comment:
Maybe rename this to `cleanupIfActiveRoom` or similar? The current name sounds like an event handler that is called when the active room is left. This will also make code clearer at the locations from where it is called.